taskcluster/fenix_taskgraph/parameters.py

```diff
@@ -7,49 +7,49 @@
 import os
 import re
 
+from mozilla_version.fenix import FenixVersion
 from six import text_type
 from taskgraph.parameters import extend_parameters_schema
 from voluptuous import All, Any, Optional, Range, Required
 
-
-BETA_SEMVER = re.compile(r'^v\d+\.\d+\.\d+-beta\.\d+$')
-RELEASE_SEMVER = re.compile(r'^v\d+\.\d+\.\d+(-rc\.\d+)?$')
-
+from .release_promotion import read_version_file
 
 extend_parameters_schema({
     Required("pull_request_number"): Any(All(int, Range(min=1)), None),
     Required("release_type"): text_type,
     Optional("shipping_phase"): Any('build', 'ship', None),
     Required("version"): text_type,
+    Required("next_version"): Any(None, text_type),
 })
 
 
 def get_decision_parameters(graph_config, parameters):
+    parameters.setdefault("release_type", "")
     head_tag = parameters["head_tag"].decode("utf-8")
-    parameters["release_type"] = resolve_release_type(head_tag)
     parameters["version"] = head_tag[1:] if head_tag else ""
 
     pr_number = os.environ.get("MOBILE_PULL_REQUEST_NUMBER", None)
     parameters["pull_request_number"] = None if pr_number is None else int(pr_number)
+    parameters.setdefault("next_version", None)
 
+    # TODO: Remove this block once github-releases are not supported anymore
     if parameters["tasks_for"] == "github-release":
-        for param_name in ("release_type", "version"):
-            if not parameters[param_name]:
-                raise ValueError(
-                    'Cannot run github-release if "{}" is not defined. Got: {}'.format(
-                        param_name, parameters[param_name]
-                    )
-                )
         parameters["target_tasks_method"] = "release"
 
-
-def resolve_release_type(head_tag):
```

Comment:
This function is not useful anymore since we can rely on https://github.com/mozilla-releng/mozilla-version. That library knows how to detect a beta version from a release one.